src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,25 +1,123 @@
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        int handLength = hands.length;
+        ArrayList<Integer> indicies = new ArrayList<Integer>();
+        int idcsLength;
+        int highestPairIdx = 0;
+        int highestScore = 0;
+        int currScore;
+        int currIdx;
+
+        for (int i = 0; i < handLength; i++){
+            if (hands[i][0].equals(hands[i][1]))
+                indicies.add(i);
+        }
+        idcsLength = indicies.size();
+
+        if (idcsLength == 0)
+            return new String[]{};
+
+        for (int i = 0; i < idcsLength; i++){
+            currIdx = indicies.get(i);
+            currScore = pointsMap.get(hands[currIdx][0]);
+            if (highestScore < currScore){
+                highestScore = currScore;
+                highestPairIdx = currIdx;
+            }
+        }
+        return hands[highestPairIdx];
     }
 
     // Extension 2
     public String[] winningThreeCardHand(String[][] hands) {
         // Implement the winningThreeCardHand logic here and return the array containing the winning hand to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        int handLength = hands.length;
+        ArrayList<Integer> threeIdcs = new ArrayList<Integer>();
+        int threeIdcsLength;
+        ArrayList<Integer> twoIdcs = new ArrayList<Integer>();
+        int twoIdcsLength;
+
+        int highestPairIdx = 0;
+        int highestScore = 0;
+        int currScore;
+        int currIdx;
+
+        for (int i = 0; i < handLength; i++){
+            if (hands[i][0].equals(hands[i][1])) {
+                if (hands[i].length == 3 && hands[i][0].equals(hands[i][2]))
+                    threeIdcs.add(i);
+                else
+                    twoIdcs.add(i);
+            }
+        }
+        threeIdcsLength = threeIdcs.size();
+        twoIdcsLength = twoIdcs.size();
+
+
+        if (threeIdcsLength == 0){
+            if (twoIdcsLength == 0)
+                return new String[]{};
+
+            for (int i = 0; i < twoIdcsLength; i++){
+                currIdx = twoIdcs.get(i);
+                currScore = pointsMap.get(hands[currIdx][0]);
+                if (highestScore < currScore){
+                    highestScore = currScore;
+                    highestPairIdx = currIdx;
+                }
+            }
+            return hands[highestPairIdx];
+        }
+
+
+        for (int i = 0; i < threeIdcsLength; i++){
+            currIdx = threeIdcs.get(i);
+            currScore = pointsMap.get(hands[currIdx][0]);
+            if (highestScore < currScore){
+                highestScore = currScore;
+                highestPairIdx = currIdx;
+            }
+        }
+        return hands[highestPairIdx];
```

Comment:
Your implementation of winningPair using a hashmap for card values is effective, and the logic in `winningPairFromArray` shows solid problem-solving skills. In winningThreeCardHand, you've effectively handled both pairs and three-of-a-kind, which demonstrates a good understanding of the task. Nice work.